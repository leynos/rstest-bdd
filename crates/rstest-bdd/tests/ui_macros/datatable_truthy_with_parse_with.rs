//! Compile-fail fixture: `#[datatable(truthy)]` and `#[datatable(parse_with = ...)]`
//! may not be combined on the same field.

use rstest_bdd_macros::DataTableRow;

fn parse_bool(value: &str) -> Result<bool, core::convert::Infallible> {
    let _ = value;
    Ok(true)
}

#[derive(Debug, DataTableRow)]
struct InvalidTruthParseField {
    #[datatable(truthy, parse_with = parse_bool)]
    value: bool,
}

fn main() {}
