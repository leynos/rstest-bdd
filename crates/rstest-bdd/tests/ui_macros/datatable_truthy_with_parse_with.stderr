error: truthy and parse_with are mutually exclusive
  --> tests/ui_macros/datatable_truthy_with_parse_with.rs:12:5
   |
12 | /     #[datatable(truthy, parse_with = parse_bool)]
13 | |     value: bool,
   | |_______________^
